gateway/challenge_verifier/verifier.go

```diff
@@ -0,0 +1,138 @@
+package challenge_verifier
+
+import (
+	"context"
+	"crypto/sha256"
```

Comment:
We should use the `minio` library over the standard library version:

```suggestion
	"github.com/minio/sha256-simd"
```